quickwit-indexing/src/actors/uploader.rs

```diff
@@ -216,32 +216,24 @@ fn create_split_metadata(split: &PackagedSplit, footer_offsets: Range<u64>) -> S
 
 fn make_publish_operation(
     index_id: String,
-    mut packaged_splits_and_metadatas: Vec<(PackagedSplit, SplitMetadata)>,
-) -> PublisherMessage {
+    packaged_splits_and_metadatas: Vec<(PackagedSplit, SplitMetadata)>,
+    checkpoint_delta_opt: Option<IndexCheckpointDelta>,
+) -> SplitUpdate {
     assert!(!packaged_splits_and_metadatas.is_empty());
     let replaced_split_ids = packaged_splits_and_metadatas
         .iter()
         .flat_map(|(split, _)| split.replaced_split_ids.clone())
         .collect::<HashSet<_>>();
-    if packaged_splits_and_metadatas.len() == 1 && replaced_split_ids.is_empty() {
-        let (mut packaged_split, split_metadata) = packaged_splits_and_metadatas.pop().unwrap();
-        assert_eq!(packaged_split.checkpoint_deltas.len(), 1);
-        let checkpoint_delta = packaged_split.checkpoint_deltas.pop().unwrap();
-        PublisherMessage::NewSplit(PublishNewSplit {
-            index_id,
-            new_split: split_metadata,
-            checkpoint_delta,
-            split_date_of_birth: packaged_split.split_date_of_birth,
-        })
-    } else {
-        PublisherMessage::ReplaceSplits(ReplaceSplits {
-            index_id,
-            new_splits: packaged_splits_and_metadatas
-                .into_iter()
-                .map(|split_and_meta| split_and_meta.1)
-                .collect_vec(),
-            replaced_split_ids: Vec::from_iter(replaced_split_ids),
-        })
+    SplitUpdate {
+        index_id,
+        new_splits: packaged_splits_and_metadatas
+            .into_iter()
+            .map(|split_and_meta| split_and_meta.1)
+            .collect_vec(),
+        replaced_split_ids: Vec::from_iter(replaced_split_ids),
+        checkpoint_delta_opt,
+        //         split_date_of_birth: packaged_split.split_date_of_birth,
```

Comment:
You're fixing this in the next PR? One `Instant` per split? Created when the first doc is added?

Reply:
This PR ditch the split date of birth for a better date that is emitted by the source.